### ISSUE TITLE
loss of internet connection by opening Nylas Mail

### ISSUE THREAD

**Issue by franco27** [nylas-mail-lives/nylas-mail#183]
Hello 

I just found a big bug!

After installation of nylas-mail-2.2.2-x64.exe.  This problem was not present.
Since 20 days, I had not of the problem. I have been forced to uninstall Nylas. and delete the Nylas user folders because an uninstall / reinstallation did not resolve this issue.

I do not know if this problem can be repeated. 
I give you step by step how this big problem happened:
 
1 / when I sent a message with an attached file of 5 MB
2/ I did not get this message error immediately but shortly thereafter ... 
but after I saw for synchronization: send of the message in the "sent" folder, this error message was displayed:
![nylas mail is offline_ error](https://user-images.githubusercontent.com/31793920/33403305-2ad56210-d560-11e7-8c64-a042fc14c966.png)

before this error message appears, apparently the message was sent (I heard the notification sound of sending),

3 / I found that my internet connection was lost 
4 / previously I went on the internet with my internet browser and my internet connection was operational (before writing and sending this message )

Now when I open Nylas:
1 / the synchronization starts ...
2 / Nylas is still trying to send this message to my "Sent" folder
3/This same error message is displayed again 
4 / I again lose my internet connection from my computer

**Comment by franco27** [comment]
I opened another messaging software.
I do not understand because I see in the folder "Sent", this message.

**Comment by franco27** [comment]
The message was sent 6 times to the "Sent" folder
 on webmail:
![2017-11-30 10_16_00-adm tools webmail __ sent](https://user-images.githubusercontent.com/31793920/33423336-c753f68a-d5b8-11e7-8b0f-c041284d25e0.png)

**This bug has corrupted my user folder. When will a tool for Nylas repair the message database?**

I deleted the .nylas-mail folder and the others user folders:
C:\Users\Francois\.nylas-mail
C:\Users\Francois\AppData\Local\NylasMail
C:\Users\Francois\AppData\Roaming\Nylas Mail
(with Windows 10)

I had to reinstall Nylas but the synchronization of my mailboxes is too long. I would have liked to have a solution to solve this problem quickly in the future.

**Comment by franco27** [comment]
Now I do not have this error message anymore and can use Nylas again.

I'm left with another problem (which I did not have previously)
1 / in the folder "Send", I have a fanciful sort!
- recent messages, are below
- old messages, are at the top
When I do not have old messages mixed with recent messages or vice versa

![2017-11-30 11_10_00-nylas mail](https://user-images.githubusercontent.com/31793920/33425560-59c14e54-d5bf-11e7-9eee-94524009792f.png)
![2017-11-30 11_06_44-nylas mail](https://user-images.githubusercontent.com/31793920/33425562-59e3c8c6-d5bf-11e7-8cc2-696671e4f9f4.png)

**Comment by franco27** [comment]
Second installation, now it's OK .
sorting the messages and the order is correct this time from the "Sent" folder

**Comment by seesemichaelj** [comment]
Thanks for the writeup @franco27. I'm having some issues following what really is the issue. Is the issue that the email was sent multiple times when you had an actual internet outage? I'm also now following how this bug corrupted your user folder (do you mean your `~/.nylas-mail` folder?)? There will likely never be a tool to do message repair for Nylas unless a developer wants to implement that.

As far as your sorting problem (which really should be a separate issue): There is a `Display conversations in descending chronological order` checkbox in the General preferences tab; I'm guessing this got checked somehow (although we can't verify due to your multiple reinstalls).

FYI there is a redacting screenshot mode if you click the menu>Developer>Toggle Screenshot Mode so you don't have to blur the details next time you take a screenshot

**Comment by franco27** [comment]
OK thank you for your answer. 
Did Nylas cause the loss of internet connection or at the time of sending I really had an internet disconnection? I can not attest ...

Nevertheless, I am formal. The problem with Nylas started with the display of the error message:
![nylas mail is offline_ error](https://user-images.githubusercontent.com/31793920/33447032-4076b3f6-d602-11e7-9eec-94e6a29a6184.png)

From that moment, I could not use Nylas anymore. Whenever I open Nylas:
1 /the message of the synchronization : starts ...
2 / then I saw : Nylas is still trying to send this message to my "Sent" folder
3/This same error message is displayed again
4 / I again lose my internet connection from my computer

I think that after this problem my user folder has been corrupted. I think about the folder .nylas-mail
This is the folder containing or is stored our mail accounts and messages?

I do not know which of the user folders has been corrupted. Still, after deleting all these user folders, I could open Nylas again without any problem.

Which makes me think of a corrupted file or files in a user folders because a simple uninstall of Nylas and a new installation without deleting the user folders did not solve these problems when Nylas started (or the error message was always displayed)
(It's after I deleted these user folders: 
C:\Users\Francois\.nylas-mail
C:\Users\Francois\AppData\Local\NylasMail
C:\Users\Francois\AppData\Roaming\Nylas Mail

To avoid a long wait for synchronization, when all is well, I would like to save the folder that contains the email accounts and all messages. Can you confirm the folder to save? (I think it's this one C:\Users\Francois\.nylas-mail)

"FYI there is a redacting screenshot mode if you click the menu>Developer>Toggle Screenshot Mode so you don't have to blur the details next time you take a screenshot"----> I just tried it does not hide the messages

If you have other questions, do not hesitate!

**Comment by seesemichaelj** [comment]
Thanks for the write up again. I'll try to take a look at this later with fresh eyes and see if I can understand what's happening here.

The `C:\Users\Francois\.nylas-mail` folder stores your config.*.json files which store your account info and auth tokens. The folder also contains the sqlite dbs for the messages/etc. So this is probably the folder you want, yes.

The screenshot mode doesnt hide message content btw. It just hides names/subjects/etc:
![image](https://user-images.githubusercontent.com/549323/33449435-c1de0272-d5d6-11e7-8fa0-49b35a3d640d.png)
![image](https://user-images.githubusercontent.com/549323/33449446-cd9785ac-d5d6-11e7-80ae-d256a103a8f9.png)
If you can't get the screenshot mode to do this, then maybe we need to write another issue.

**Comment by franco27** [comment]
this is the theme I had installed that is at fault 

no problem now

**Comment by dweremeichik** [comment]
@franco27 which theme?

**Comment by seesemichaelj** [comment]
@dweremeichik I'm also thinking his theme was referring to the screenshot mode issue, not the original problem with this issue. is this correct @franco27?

**Comment by dweremeichik** [comment]
@seesemichaelj yeah, that's what I gathered. Would be good to know which one. Also, I find it hard to believe that Nylas could directly kill your internet, however I do believe that it could be giving your system the good old electron hug of death. If you could, open your system monitor and watch it while this happens, both for memory and network traffic, that would be a start. Also it would be good to know how much RAM your system has.

**Comment by dweremeichik** [comment]
@franco27 please see my previous comment, but also: does your network icon in the bottom right of your screen actually change? The offline notification in Nylas can be a bit buggy at times.

**Comment by franco27** [comment]
Sorry for the late response with the jet lag (I live in France and notifications of your messages have arrived this night)

the screenshot problem is related to the theme I had installed (theme firefox) (it is not installed at the Nylas installation)

> @dweremeichik I'm also thinking his theme was referring to the screenshot mode issue, not the original problem with this issue. is this correct 
yes seesemichaelj 

This problem causes the loss of the Internet connection of the computer and not only of Nylas.
I would be surprised that you may not be able to reproduce this problem.

My computer has a good configuration: judged rather ...
QuadCore Intel Core i5-3570, 3800 MHz (38 x 100)
8153 Mo  (DDR3 SDRAM) - 2x 4 Go DDR3-1333 DDR3 SDRAM  (9-9-9-24 @ 666 MHz)  (8-8-8-22 @ 592 MHz)  (7-7-7-19 @ 518 MHz)  (6-6-6-16 @ 444 MHz)
Windows 10 x64bits

I had exactly the same problem with Mailspring. Only after a restart of Mailspring, the problem has been resolved and the error message is no longer displayed. Similarly, this error message has been accompanied by a loss of my Internet connection on the computer. (While sending the message / during sending the message with file attached)

With Nylas Mails Live, this problem persists at restart and always with this schema:
1 /the message of the synchronization : starts ...
2 / then I saw : Nylas is still trying to send this message to my "Sent" folder
3/This same error message is displayed again
4 / I again lose my internet connection from my computer
5 / and after, I could observe a change in the status of the "Network" icon
![2017-12-14 13_25_41-](https://user-images.githubusercontent.com/31793920/33992541-563837ba-e0d3-11e7-887c-38b878e2ec29.png)
---->
![2017-12-14 13_31_46-](https://user-images.githubusercontent.com/31793920/33992556-63af78c2-e0d3-11e7-9b0a-c705e08a4ee5.png)

**Comment by franco27** [comment]
the change of status of the icon "Network" is not immediate of course. that's when I try to go on the internet or using the internet, and insisting that I then see the status of the Network be changed

I am connected in Ethernet 
I assure you that after this problem and opening of Nylas Mail Lives, after displaying the error message
![nylas mail is offline_ error](https://user-images.githubusercontent.com/31793920/33993058-455a7212-e0d5-11e7-95e6-c693377e70ff.png)

my internet connection becomes inactive.
I am in ADSL (I can not benefit from an internet fiber connection)

to find my active internet connection, I was forced to close Nylas and disable / enable my network card (maybe disconnecting the Ethernet cable would have corrected this problem)

maybe the size of the attached file (5 MB and larger size) is a problem for Nylas
(I do not remember sending an attached file smaller than this size)

no problem in sending a conventional message without attached file
and also with sending a message, by pasting an image of the clipboard into the body of the message
 
currently, I'm trying Nylas and Mailspring. I inform you, when Nylas is open, Mailspring is closed completely (I hear that there is no icon in the notification area)
and vice versa

**Comment by franco27** [comment]
I hope I've answered all of your questions. Thank you

**Comment by seesemichaelj** [comment]
Unfortunately, we haven't had any of these reports by other users, and as far as the source goes, there's nothing that prevents larger than 5MB file attachments.

With this, I'm thinking there's something else at play. Can you try sending similar emails using other email clients (i.e. Windows Mail, Outlook, Thunderbird) and see if they have similar issues? Can you upload a file to Dropbox or Google Drive to see if network goes out? Might be interesting if it's an Electron issue.

**Comment by seesemichaelj** [comment]
Can you also try opening the task manager (Ctrl+shift+esc), click advanced on the bottom left, then go to the performance tab. Watch the network usage as you try to send an email with an attachment. Would be interesting if the network card is just getting bombarded and can't take any more until the traffic goes through. I.e. does the network usage go up as you send and then just cut off when you lose network, or does it stay going even though it says you lost network

**Comment by franco27** [comment]
I have no problem sending attachments with other email software
I use google drive regularly, no problem (no problem sending)

while sending the message with the attached file, no downloads at this time

second hypothesis, we must check 

the message was sent, the recipient received it but problem after

**Comment by franco27** [comment]
I tried to deepen the question. I have done of the tests.

Indeed, Nylas does not disable the internet connection but takes too much network traffic and blocks other applications. (prevents other applications from accessing the internet)
At a given time , I have when even had:
![2017-12-14 13_31_46-](https://user-images.githubusercontent.com/31793920/34018697-6e8faf34-e12b-11e7-8536-9af4c30a90e1.png)

1/ First Test 
from my @gmail address, I sent a message with an attached file (5MB) to my other @gmx mail address
----> no problem

2/ second test 
2.1) from my @antdownloadmanager.com address (Roundcube Webmail 1.3.3), I sent a message with an attached file (5MB) to my other @gmail mail address
2.2) The message has been sent
----> **after the problems started .**...
2.3) During the copy of the message in the folder "sent", this error message was displayed:
![nylas mail is offline_ error](https://user-images.githubusercontent.com/31793920/34019189-f72a8d8a-e12d-11e7-8e07-6c4d179d6699.png)
and
![2017-12-14 17_44_47-moniteur de ressources](https://user-images.githubusercontent.com/31793920/34019249-3e49eada-e12e-11e7-8039-409d15903f0b.png)
At this moment, I can not go on the internet with Firefox or Vivaldi

2.4) After a wait of 5 to 10 minutes, I finally have my active internet connection again and can use it

2.5) this message is always displayed  , I did not manage to remove it
![nylas mail is offline_ error](https://user-images.githubusercontent.com/31793920/34019480-77c4d382-e12f-11e7-9474-d9aeb4faf0cf.png)

2.6) I close Nylas and restart, 
1 /the message of the synchronization : starts ...
2 / then I saw : Nylas is still trying to send this message to my "Sent" folder
![2017-12-14 17_49_20-nylas mail](https://user-images.githubusercontent.com/31793920/34019591-0701b34e-e130-11e7-9407-3cb8a3900e9e.png)

3/This same error message is displayed again
4 / I again lose my internet connection from my computer
I see 
![2017-12-14 18_01_39-gestionnaire des taches](https://user-images.githubusercontent.com/31793920/34019647-4b28ca94-e130-11e7-8fd9-16324af33a9a.png)
![2017-12-14 17_47_18-moniteur de ressources](https://user-images.githubusercontent.com/31793920/34019648-4b481a0c-e130-11e7-8ecf-7a1e11218f79.png)

I'm waiting again (this time the wait to go on the internet is shorter)

After that my internet connection is operational, there is a slowdown in Nylas . Messages are always displayed:
![2017-12-14 17_49_20-nylas mail](https://user-images.githubusercontent.com/31793920/34019928-7d14e4a6-e131-11e7-9cc3-9fa4cd1f951b.png)
![nylas mail is offline_ error](https://user-images.githubusercontent.com/31793920/34019929-7d369a4c-e131-11e7-95d0-26b9634c9573.png)

I can hardly use Nylas, 
The receipt of a new message and after clicking it caused the message to disappear:
![2017-12-14 17_49_20-nylas mail](https://user-images.githubusercontent.com/31793920/34020071-5e6ce548-e132-11e7-9016-c825e8692954.png)
the other message is always displayed and is persistent: "Nylas offline"

I think that the reception of a new message made unblock this situation.
because the first time, he did not want to disappear and each restart "loss of internet connection".
or
when i saw that my internet connection was inactive, i closed too early nylas (i did not wait long enough)

2.7) I close Nylas and restart, 
this time Nylas is working properly again

in the "Sent" folder, I see a dozen messages for @ant (Roundcube Webmail ) for a single message received on @gmail

**Comment by seesemichaelj** [comment]
Great diagnosis! Just what I thought; looks like your bandwidth is getting hogged by the upload/download from Nylas. I'm thinking this is just how the download library being used is working. I don't know if we can do this, but the only potential solution we could maybe provide is to allow some throttling of the speed in the configuration. What are your thoughts @dweremeichik?

**Comment by franco27** [comment]
I did a third test
from my @antdownloadmanager.com address (Roundcube Webmail 1.3.3), I sent a message with an attached file (5MB) to my other @gmail mail address

this time I used another email software Thunderbird  ---> no problem

**Comment by dweremeichik** [comment]
@seesemichaelj I'm not sure that's even possible. Perhaps it is worth a look. @franco27 if you do a speed test, what's your up and down?

**Comment by seesemichaelj** [comment]
And if it was, not sure there is anyone that would have/take the time to implement it :/

**Comment by lamixer** [comment]
On Fri, 15 Dec, 2017 at 12:09 AM, franco27 <notifications@github.com> 
wrote:

> in the "Sent" folder, I see a dozen messages for @Ant (Roundcube 
> Webmail ) for a single message received on @gmail

Does this sound like Nylas Mail is mistakenly copying the sent message 
(and attachement) 12 times, thus an error in mailbox/message handling 
in Nylas Mail?

**Comment by franco27** [comment]
yes, Nylas Mail copied the message sent, a dozen times (same message) in the "Sent" folder

for @Ant (Roundcube Webmail ) , error in email management

![the global broadband speed test](https://user-images.githubusercontent.com/31793920/34041469-67466c14-e198-11e7-8ebe-33e87a7c54dd.png)
